src/hdf5_interface.F90

```diff
@@ -1320,7 +1331,7 @@ subroutine read_attribute_string(buffer, obj_id, name)
 
     ! Allocate a C char array to get string
     n = attribute_typesize(obj_id, to_c_string(name))
-    allocate(buffer_(n))
+    allocate(buffer_(n+1))
```

Comment:
Yeah, I noticed this one too, but I actually have a better fix for it (and the other HDF5-related errors). Do you mind reverting this change and then I'll put in another PR that fixes the remaining leaks?